platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/StateSigningTests.java

```diff
@@ -482,56 +462,47 @@ void signaturesInvalidDueToZeroWeightTest(final boolean evenWeighting) {
 
         final int nodeCount = random.nextInt(10, 20);
 
-        final AddressBook addressBook = RandomAddressBookBuilder.create(random)
+        final Roster roster = RandomRosterBuilder.create(random)
                 .withWeightDistributionStrategy(
-                        evenWeighting
-                                ? RandomAddressBookBuilder.WeightDistributionStrategy.BALANCED
-                                : RandomAddressBookBuilder.WeightDistributionStrategy.GAUSSIAN)
+                        evenWeighting ? WeightDistributionStrategy.BALANCED : WeightDistributionStrategy.GAUSSIAN)
                 .withSize(nodeCount)
                 .build();
 
         // set node to zero weight
-        final NodeId nodeWithZeroWeight = addressBook.getNodeId(0);
-        addressBook.updateWeight(nodeWithZeroWeight, 0);
+        final RosterEntry zeroWeightEntry =
+                roster.rosterEntries().getFirst().copyBuilder().weight(0L).build();
+        RosterTestUtils.replaceNode(roster, zeroWeightEntry);
```

Comment:
Same as above: Roster is an immutable object by definition/specification. So we can no longer update it in place. We can only create a new instance with whatever updates we want to apply.

This may invalidate certain tests and use case. Or one needs to be creative to see how to re-implement them w/o losing the immutability property of the Roster.